[T12-1] Hotel Management System

sreycodes opened the pull request.

sreycodes commented:
@tejas2805 @nerrons @ZhangCX10032

bqnguyen94 commented on README.adoc:
```diff
@@ -1,32 +1,19 @@
 https://travis-ci.org/se-edu/addressbook-level4[image:https://travis-ci.org/se-edu/addressbook-level4.svg?branch=master[Build Status]]
 https://ci.appveyor.com/project/damithc/addressbook-level4[image:https://ci.appveyor.com/api/projects/status/3boko2x2vr5cc3w2?svg=true[Build status]]
 https://coveralls.io/github/se-edu/addressbook-level4?branch=master[image:https://coveralls.io/repos/github/se-edu/addressbook-level4/badge.svg?branch=master[Coverage Status]]
 https://www.codacy.com/app/damith/addressbook-level4?utm_source=github.com&utm_medium=referral&utm_content=se-edu/addressbook-level4&utm_campaign=Badge_Grade[image:https://api.codacy.com/project/badge/Grade/fc0b7775cf7f4fdeaf08776f3d8e364a[Codacy Badge]]
```
Please setup CIs and update these badges

bqnguyen94 commented on README.adoc:
```diff
@@ -1,32 +1,19 @@
-* <<LearningOutcomes#, Learning Outcomes>>
 * <<AboutUs#, About Us>>
 * <<ContactUs#, Contact Us>>
 
```
Don't forget to add `se-edu` to `Acknowledgements` below

bqnguyen94 commented on README.adoc:
```diff
@@ -1,32 +1,19 @@
-endif::[]
-
-ifndef::env-github[]
-image::images/Ui.png[width="600"]
```
The picture doesn't have to be a screenshot of the actual product. A mockup is totally fine as well so don;t forget to add it in.

bqnguyen94 reviewed (commented):
I see that you guys have started working on the product 👍 
There are still places in `User Guide` that uses `Address Book` instead of `Hotel Management System`, please update them soon. Also, it should be easier to update `Developer Guide` as you code along as well, starting with updating the name of the app

sreycodes commented:
> I see that you guys have started working on the product 👍
> There are still places in `User Guide` that uses `Address Book` instead of `Hotel Management System`, please update them soon. Also, it should be easier to update `Developer Guide` as you code along as well, starting with updating the name of the app

Regarding my feedback for week 6 - I feel there is a discrepancy based on what the bot has reported. Can you check it @bqnguyen94 https://github.com/cs2103-ay1819s2-t12-1/main/issues/8

Tejas2805 committed:
Corrected errors to please checkStyle.

Tejas2805 committed:
Corrected errors for checkStyle.

Tejas2805 committed:
Corrected errors.

Tejas2805 committed:
Made changes to checking for DateOfBirth.

Tejas2805 committed:
Corrected errors.

Tejas2805 committed:
Corrected tests.

Tejas2805 committed:
Merge branch 'master' into master

Tejas2805 committed:
Fixed bugs founds in Dry Run.
Bug 1: Edited customer can be made same as existing customer
Bug 2: Edit reservation example command
Bug 3: Customer indices for other Users shows exception if invalid.

Tejas2805 committed:
Merge branch 'master' into Bug-fixes

Tejas2805 committed:
Added tests for Reservation.
Added tests for FindCommands.
Added tests for ParserCommands.

Tejas2805 committed:
Added tests for HotelParser.

Tejas2805 committed:
Merge branch 'Bug-fixes' of https://github.com/Tejas2805/main into Bug-fixes

* 'Bug-fixes' of https://github.com/Tejas2805/main:
  Removed unneccesary file
  Fixed tests
  Fixed more checkstyle stuff
  Fixed error
  Fixed room type errors
  Fixed declaration order
  Fixed bug regarding booking and reservation
  Can edit types
  Can add service types and room types
  ServiceType and RoomType is now customizable

# Conflicts:
#	src/test/data/JsonSerializableHotelManagementSystemTest/typicalCustomersHotelManagementSystem.json
#	src/test/java/seedu/hms/logic/parser/HotelManagementSystemParserTest.java

ZhangCX10032 committed:
panel will be scrolled to the newly added item now

ZhangCX10032 committed:
the service type and room type panel will not be scrolled

ZhangCX10032 committed:
fix unit test for adding commands

ZhangCX10032 committed:
fix test

ZhangCX10032 committed:
fix some UI tests

ZhangCX10032 committed:
improve style

Tejas2805 committed:
Correct errors for Travis.

Tejas2805 committed:
Corrected errors for checkStyle and codacy.
Made tests for reservation run.

Tejas2805 committed:
Corrected errors for checkStyle.

Tejas2805 committed:
Corrected codacy errors.

Tejas2805 committed:
Corrected errors to please checkStyle.

Tejas2805 committed:
Corrected errors for travis.

ZhangCX10032 committed:
Fix the date range bug
  now a valid date should be equals to or after today and with in one year after current date

ZhangCX10032 committed:
Update user guide

ZhangCX10032 committed:
Update user guide

Tejas2805 committed:
Merge pull request #217 from Tejas2805/Bug-fixes

Bug fixes.

Tejas2805 committed:
Merge branch 'master' into ServiceTypeAndRoomType

Tejas2805 committed:
Added tests.
Fixed original tests.

Tejas2805 committed:
Merge branch 'master' into Bug-fixes

Tejas2805 committed:
Corrected errors.

Tejas2805 committed:
Corrected errors.

Tejas2805 committed:
Correct errors.

Tejas2805 committed:
Corrected errors.

Tejas2805 committed:
Corrected errors.

Tejas2805 committed:
Corrected errors.

Tejas2805 committed:
Corrected original tests.

ZhangCX10032 committed:
Add check for 0 in bill panel so that it will not print service/room types with 0 amount
remove some System.out.println() codes

Tejas2805 committed:
Added tests for deleteReservationCommand.

Tejas2805 committed:
Added parser tests for GenerateBillCommands.

Tejas2805 committed:
Corrected errors for checkStyle.
Added parser tests for SwitchTabCommand.

Tejas2805 committed:
Corrected errors for checkStyle.

ZhangCX10032 committed:
Merge pull request #218 from ZhangCX10032/ServiceTypeAndRoomType

Date Range bug fix, new panel feature and some UI test fix

ZhangCX10032 committed:
Merge branch 'ServiceTypeAndRoomType' of https://github.com/ZhangCX10032/main into ServiceTypeAndRoomType

ZhangCX10032 committed:
Handle numberFormatException in switch tab command parser

Tejas2805 committed:
Added tests to SwitchTabCommandParser

Tejas2805 committed:
Merge branch 'master' into Bug-fixes

Tejas2805 committed:
Added tests for HotelManagementSystem.

Tejas2805 committed:
Corrected UG commands.

Tejas2805 committed:
Merge branch 'Bug-fixes' of https://github.com/Tejas2805/main into Bug-fixes

* 'Bug-fixes' of https://github.com/Tejas2805/main:
  Update user guide
  Update user guide
  Fix the date range bug   now a valid date should be equals to or after today and with in one year after current date
  improve style
  fix some UI tests
  fix test
  fix unit test for adding commands
  the service type and room type panel will not be scrolled
  panel will be scrolled to the newly added item now

# Conflicts:
#	docs/UserGuide.adoc

Tejas2805 committed:
Corrected errors.

Tejas2805 committed:
Corrected errors.

Tejas2805 committed:
Corrected tests.

Tejas2805 committed:
Added more tests overall.

Tejas2805 committed:
Corrected errors for checkStyle.

Tejas2805 committed:
Added more tests.

Tejas2805 committed:
Corrected errors.

ZhangCX10032 committed:
Implement tests for dateRange and timeRange

ZhangCX10032 committed:
remove some test cases to please codacy

ZhangCX10032 committed:
improve style

Tejas2805 committed:
Added tests for GenerateBillForReservation as ReservationWithDatePredicate was activated

Tejas2805 committed:
Corrected errors to plesase checkStyle.

Tejas2805 committed:
Corrected errors.

sreycodes committed:
Merge pull request #219 from Tejas2805/Bug-fixes

Added tests to improve coverage.

sreycodes committed:
Merge branch 'master' into ServiceTypeAndRoomType

ZhangCX10032 committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-t12-1/main into ServiceTypeAndRoomType

ZhangCX10032 committed:
Merge branch 'ServiceTypeAndRoomType' of https://github.com/ZhangCX10032/main into ServiceTypeAndRoomType

Tejas2805 committed:
Made new branch and merged

Tejas2805 committed:
Merge remote-tracking branch 'origin/master' into Bug-fixes

ZhangCX10032 committed:
adjust test case

sreycodes committed:
Merge pull request #221 from ZhangCX10032/ServiceTypeAndRoomType

Implementing tests, Switch tab command bug fix and bill UI fix

sreycodes committed:
No duplicates in service and room type list

Tejas2805 committed:
Merge branch 'master' into DeleteServiceAndRoomType

Tejas2805 committed:
Merge branch 'Bug-fixes' of https://github.com/Tejas2805/main into DeleteServiceAndRoomType

* 'Bug-fixes' of https://github.com/Tejas2805/main: (43 commits)
  Corrected errors.
  Corrected errors to plesase checkStyle.
  Added tests for GenerateBillForReservation as ReservationWithDatePredicate was activated
  Corrected errors.
  Added more tests.
  Corrected errors for checkStyle.
  Added more tests overall.
  Corrected tests.
  Corrected errors.
  Corrected errors.
  Corrected UG commands.
  Added tests for HotelManagementSystem.
  Added tests to SwitchTabCommandParser
  Corrected errors for checkStyle.
  Corrected errors for checkStyle. Added parser tests for SwitchTabCommand.
  Added parser tests for GenerateBillCommands.
  Added tests for deleteReservationCommand.
  Corrected original tests.
  Corrected errors.
  Corrected errors.
  ...

# Conflicts:
#	src/main/java/seedu/hms/logic/commands/FindReservationCommand.java
#	src/main/java/seedu/hms/logic/parser/ParserUtil.java
#	src/test/java/seedu/hms/logic/commands/EditReservationCommandTest.java
#	src/test/java/seedu/hms/logic/parser/HotelManagementSystemParserTest.java

sreycodes committed:
Merge branch 'master' into NoDupServiceRoomTypes

Tejas2805 committed:
Corrected errors for checkStyle.

Tejas2805 committed:
Corrected errors.

Tejas2805 committed:
Corrected errors.

Tejas2805 committed:
Corrected errors.

nerrons committed:
Add a primitive text report for stats command

sreycodes committed:
Manual testing for service and room type remaining

nerrons committed:
Abstract out StatsItems

sreycodes committed:
Finished manual testing and use cases

nerrons committed:
Ensure alignment for longest key

ZhangCX10032 committed:
undo is working for service type and room type commands now

ZhangCX10032 committed:
Fix the bug booking can be added in from time to the same time

ZhangCX10032 committed:
BillPanel will show message if there is no booking or reservation available for the selected customer

sreycodes committed:
Merge pull request #223 from Tejas2805/DeleteServiceAndRoomType

Delete service and room type

sreycodes committed:
Merge branch 'master' into ServiceTypeAndRoomType

sreycodes committed:
Merge pull request #226 from ZhangCX10032/ServiceTypeAndRoomType

implement undo and redo for serviceType & roomType and fix some bugs

sreycodes committed:
Merge branch 'master' into NoDupServiceRoomTypes

Tejas2805 committed:
Merge branch 'master' into DeleteServiceAndRoomType

Tejas2805 committed:
Tiny bugs like space errors in command examples fixed.
UG updated. DG updated.

sreycodes committed:
Merge pull request #227 from Tejas2805/DeleteServiceAndRoomType

UG and DG updated.

sreycodes committed:
DG Stuff

sreycodes committed:
Now edit also checks for uniqueness

sreycodes committed:
Merge branch 'NoDupServiceRoomTypes' of https://github.com/sreycodes/main into NoDupServiceRoomTypes

sreycodes committed:
Merge branch 'master' into NoDupServiceRoomTypes

sreycodes committed:
Merge pull request #222 from sreycodes/NoDupServiceRoomTypes

No duplicates in service and room type list

Tejas2805 committed:
Implemented ClearRoomTypeCommand and ClearServiceTypeCommand.

Tejas2805 committed:
Corrected errors.

Tejas2805 committed:
Corrected errors.

Tejas2805 committed:
Merge branch 'master' into master

ZhangCX10032 committed:
Fix dateRange withinMethod only comparing dates and not including time
Correct message for full reservation

ZhangCX10032 committed:
fix room full check method

ZhangCX10032 committed:
Merge branch 'ServiceTypeAndRoomType' of https://github.com/ZhangCX10032/main into ServiceTypeAndRoomType

ZhangCX10032 committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-t12-1/main into ServiceTypeAndRoomType

nerrons committed:
Merge branch 'master' into add-stats

nerrons committed:
Add docs for stats

nerrons committed:
Add layout for the stats window

sreycodes committed:
Merge pull request #228 from Tejas2805/master

Implemented ClearRoomTypeCommand and ClearServiceTypeCommand.

Tejas2805 committed:
Added tests to increase coverage.
Added tests to DateOfBirth and SwitchTabCommand.
Implemented new tests for BookingCard and ReservationCard.

Tejas2805 committed:
Merge branch 'master' into master

ZhangCX10032 committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-t12-1/main into ServiceTypeAndRoomType

sreycodes committed:
Merging

sreycodes committed:
Merging

sreycodes committed:
Implement interface properly

sreycodes committed:
Merge pull request #230 from Tejas2805/master

Added tests to increase coverage.

sreycodes committed:
Merge branch 'master' into ServiceTypeAndRoomType

sreycodes committed:
Merge pull request #229 from ZhangCX10032/ServiceTypeAndRoomType

Fix reservation full check and room full message

Tejas2805 committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-t12-1/main

* 'master' of https://github.com/cs2103-ay1819s2-t12-1/main:

# Conflicts:

Improved Dob and Dob tests. Improved customer display function.

Tejas2805 committed:
Merge branch 'master' of https://github.com/Tejas2805/main

* 'master' of https://github.com/Tejas2805/main:

Tejas2805 committed:
Merge branch 'master' into master

Tejas2805 committed:
Merge pull request #232 from Tejas2805/master

Corrected DOB current date.

Tejas2805 committed:
Made UI for RoomTypeListCardCard and ServiceTypeListCard easier to understand and find by adding indexes.

Tejas2805 committed:
Merge branch 'master' of https://github.com/Tejas2805/main

* 'master' of https://github.com/Tejas2805/main:
  fix room full check method
  Fix dateRange withinMethod only comparing dates and not including time Correct message for full reservation

Tejas2805 committed:
Merge pull request #233 from Tejas2805/master

Added indexes for RoomTypeListCard and ServiceTypeListCard

Tejas2805 committed:
Added tests for UI to increase coverage.

Tejas2805 committed:
Merge branch 'master' into master

Tejas2805 committed:
Merge pull request #234 from Tejas2805/master

Added tests for UI to increase coverage.

Tejas2805 committed:
Updated UG for final stage. Only example pictures are left.

Tejas2805 committed:
Merge branch 'master' into master

nerrons committed:
Merge branch 'master' into add-stats

nerrons committed:
Make StatsItem store the map so it can be reused

sreycodes committed:
Merge branch 'master' into DevGuide

sreycodes committed:
Merge pull request #235 from Tejas2805/master

Updated UG for final stage.

nerrons committed:
Add charts

nerrons committed:
Add a filter mechanism for stats

nerrons committed:
Add options for what statitems to show

ZhangCX10032 committed:
implement booking list panel test

ZhangCX10032 committed:
comment out unused variables

nerrons committed:
Handle wrong commands

ZhangCX10032 committed:
refactor bookingListPanel's and reservationListPanel's constructor
implement test for reservationListPanel

nerrons committed:
Please codacy and checkstyle

nerrons committed:
Please codacy and checkstyle again

nerrons committed:
Merge branch 'master' into add-stats

nerrons committed:
Fix tests

ZhangCX10032 committed:
fix numOfCustomers in booking and add comment

ZhangCX10032 committed:
remove unused imports

ZhangCX10032 committed:
reduce the size of performance test

sreycodes committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-t12-1/main into DevGuide

sreycodes committed:
Merge pull request #236 from ZhangCX10032/ServiceTypeAndRoomType

implement test

sreycodes committed:
Merge branch 'master' into add-stats

ZhangCX10032 committed:
Fix service full check bug

Signed-off-by: ZhangCX10032 <549191059@qq.com>

ZhangCX10032 committed:
updated UIComponentClassDiagram

sreycodes committed:
Adding my stuff

sreycodes committed:
Add stuff

sreycodes committed:
Merge branch 'DevGuide' of https://github.com/sreycodes/main into DevGuide

Tejas2805 committed:
Updated all docs - PPP, User Guide and Developer Guide.

Tejas2805 committed:
Merge branch 'master' into master

Tejas2805 committed:
Merge pull request #238 from Tejas2805/master

Updated all docs - PPP, User Guide and Developer Guide.

ZhangCX10032 committed:
fix reservation full check bug

ZhangCX10032 committed:
update UI picture
update UI component diagram
update DeveloperGuide
add PPP for Zhang Chenxi

ZhangCX10032 committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-t12-1/main into ServiceTypeAndRoomType

# Conflicts:
#	docs/diagrams/UiComponentClassDiagram.pptx
#	docs/images/UiClassDiagram.png

Tejas2805 committed:
Add intro to the PPP

Tejas2805 committed:
Merge branch 'master' of https://github.com/Tejas2805/main

* 'master' of https://github.com/Tejas2805/main:
  reduce the size of performance test
  remove unused imports
  fix numOfCustomers in booking and add comment
  refactor bookingListPanel's and reservationListPanel's constructor implement test for reservationListPanel
  comment out unused variables
  implement booking list panel test

Tejas2805 committed:
Merge branch 'master' into master

Tejas2805 committed:
Merge pull request #239 from Tejas2805/master

Intro to ppp

sreycodes committed:
Finish my part

sreycodes committed:
Merge branch 'master' into add-stats

sreycodes committed:
Merge branch 'master' into ServiceTypeAndRoomType

sreycodes committed:
Merge branch 'master' into DevGuide

sreycodes committed:
Merge pull request #225 from nerrons/add-stats

Add stats

sreycodes committed:
Merge branch 'master' into DevGuide

sreycodes committed:
Merge branch 'master' into ServiceTypeAndRoomType

sreycodes committed:
Removed unnecessary html file

sreycodes committed:
Merge branch 'DevGuide' of https://github.com/sreycodes/main into DevGuide

Tejas2805 committed:
Merge pull request #231 from sreycodes/DevGuide

Dev guide

sreycodes committed:
Merge branch 'master' into ServiceTypeAndRoomType

sreycodes committed:
PPP

sreycodes committed:
Merge branch 'master' into DevGuide

ZhangCX10032 committed:
update user guide
update PPP

ZhangCX10032 committed:
Merge branch 'ServiceTypeAndRoomType' of https://github.com/ZhangCX10032/main into ServiceTypeAndRoomType

ZhangCX10032 committed:
update PPP

ZhangCX10032 committed:
Merge pull request #240 from sreycodes/DevGuide

PPP

ZhangCX10032 committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-t12-1/main into ServiceTypeAndRoomType

ZhangCX10032 committed:
Merge pull request #237 from ZhangCX10032/ServiceTypeAndRoomType

Bug fix and Doc update

sreycodes committed:
Added images for misc command

Tejas2805 committed:
Updated UG completely with all relevant pics.

Tejas2805 committed:
Merge branch 'master' of https://github.com/Tejas2805/main

* 'master' of https://github.com/Tejas2805/main:

Tejas2805 committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-t12-1/main

* 'master' of https://github.com/cs2103-ay1819s2-t12-1/main: (29 commits)
  update PPP
  update user guide update PPP
  PPP
  Removed unnecessary html file
  Finish my part
  update UI picture update UI component diagram update DeveloperGuide add PPP for Zhang Chenxi
  fix reservation full check bug
  Add stuff
  Adding my stuff
  updated UIComponentClassDiagram
  Fix service full check bug
  Fix tests
  Please codacy and checkstyle again
  Please codacy and checkstyle
  Handle wrong commands
  Add options for what statitems to show
  Add a filter mechanism for stats
  Add charts
  Make StatsItem store the map so it can be reused
  Implement interface properly
  ...

# Conflicts:
#	docs/DeveloperGuide.adoc

sreycodes committed:
Added image for undo

sreycodes committed:
Add images

sreycodes committed:
Merge branch 'master' into master

sreycodes committed:
merge

Tejas2805 committed:
Merge pull request #242 from sreycodes/master

Add images to User Guide

ZhangCX10032 committed:
Set theme jekyll-theme-slate

sreycodes committed:
Change sreycodes PPP link

ZhangCX10032 committed:
Update ZhangChenxi.adoc

sreycodes committed:
Change image position

sreycodes committed:
Merge branch 'master' into master

ZhangCX10032 committed:
update PPP and about us page

sreycodes committed:
Merge pull request #241 from Tejas2805/master

Updated UG with diagrams.

sreycodes committed:
Updating readme

sreycodes committed:
Make toc to the side

sreycodes committed:
Merge branch 'master' into Readme

ZhangCX10032 committed:
add diagram in developer guide

sreycodes committed:
Fixed typo

ZhangCX10032 committed:
add diagram in developer guide

sreycodes committed:
fix typo

ZhangCX10032 committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-t12-1/main into ServiceTypeAndRoomType

sreycodes committed:
Fix formatting issue

nerrons committed:
Add Stats to DG

nerrons committed:
Add Stats to UG

ZhangCX10032 committed:
adjust size of the diagram

Tejas2805 committed:
Final update PPP

Tejas2805 committed:
Merge branch 'master' of https://github.com/Tejas2805/main

* 'master' of https://github.com/Tejas2805/main:
  Change image position
  Update ZhangChenxi.adoc
  Change sreycodes PPP link
  Set theme jekyll-theme-slate
  Add images
  Added image for undo
  Added images for misc command

Tejas2805 committed:
Merge branch 'master' into master

sreycodes committed:
Merge branch 'Readme' of https://github.com/sreycodes/main into Readme

sreycodes committed:
Fixed all typos

sreycodes committed:
Merge pull request #246 from Tejas2805/master

Updated PPP.

sreycodes committed:
Merge branch 'master' into Readme

ZhangCX10032 committed:
update DG

ZhangCX10032 committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-t12-1/main into ServiceTypeAndRoomType

Tejas2805 committed:
Updated ppp

sreycodes committed:
Merge pull request #243 from sreycodes/Readme

Readme

Tejas2805 committed:
Merge branch 'master' of https://github.com/Tejas2805/main

* 'master' of https://github.com/Tejas2805/main:

sreycodes committed:
Merge branch 'master' into ServiceTypeAndRoomType

sreycodes committed:
Merge branch 'master' into shaoqian-finals

sreycodes committed:
Update DeveloperGuide.adoc

sreycodes committed:
Merge pull request #244 from ZhangCX10032/ServiceTypeAndRoomType

add diagram in developer guide update about us  doc

sreycodes committed:
Merge branch 'master' into shaoqian-finals

nerrons committed:
Add Portfolio

nerrons committed:
Merge branch 'shaoqian-finals' of github.com:nerrons/main into shaoqian-finals

Tejas2805 committed:
ppp

Tejas2805 committed:
Merge branch 'master' into master

nerrons committed:
Merge pull request #245 from nerrons/shaoqian-finals

Add Stats to DG and UG

Tejas2805 committed:
Merge branch 'master' into master

Tejas2805 committed:
Merge pull request #247 from Tejas2805/master

PPP

sreycodes committed:
Update DeveloperGuide.adoc

nerrons committed:
fix

nerrons committed:
Merge branch 'master' into shaoqian-finals

nerrons committed:
Merge pull request #248 from nerrons/shaoqian-finals

fix

sreycodes committed:
Update AboutUs.adoc

Tejas2805 committed:
ppp

Tejas2805 committed:
Merge branch 'master' into master

sreycodes committed:
Update AboutUs.adoc

Tejas2805 committed:
Merge branch 'master' into master

Tejas2805 committed:
Merge pull request #249 from Tejas2805/master

ppp
